### PR TITLE
RAP-4359

### PR DESCRIPTION
**Pull request by scarabus**
Index in a separate thread

### CODE REVIEW

**Comment by jonathan-major** [comment]
lets get this done today. done == passing build and tested. @scarabus @jeremy-flieder

**Comment by scarabus** [comment]
Test cases now pass. Problem was that indexing is now asynchronous, so there was a race condition.

File: Libs/RaptureCore/src/main/java/rapture/repo/meta/AbstractMetaBasedRepo.java [hunk 1]
```diff
@@ -89,11 +89,16 @@ public MH getMetaHandler() {
 
     @Override
     public void setIndexProducer(IndexProducer producer) {
-        if (this.producer != null && log.isDebugEnabled()) {
-            // THIS APPEARS TO BE BENIGN AS WE ARE PRETTY MUCH SETTING THE SAME
-            // THING UP SO NOT FIXING FOR NOW
-            log.debug("Producer is already set to " + this.producer + " am overwriting!!!");
-            log.debug("New is " + producer.toString() + ", Old is " + this.producer.toString());
+        if (this.producer != null) {
+            System.out.println(this.producer.toString());
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/repo/meta/AbstractMetaBasedRepo.java]: change this

File: Libs/RaptureCore/src/main/java/rapture/repo/meta/handler/AbstractMetaHandler.java [hunk 2]
```diff
@@ -38,515 +48,473 @@
...
+                    public boolean visit(String key, String value) {
+                        log.info("Index rebuild, rebuilding " + key);
+                        DocumentMetadata mdLatest = getLatestMeta(key);
+                        indexHandler.get().addedRecord(key, value, mdLatest);
+                        return true;
+                    }
+                });
+            }
+        };
+        thread.start();
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/repo/meta/handler/AbstractMetaHandler.java]: how does the thread get destroyed?

Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/repo/meta/handler/AbstractMetaHandler.java]: change info to trace

**Pull request merged**